src/wallet/db.h

```diff
@@ -111,6 +117,12 @@ class CWalletDBWrapper
      */
     std::string GetName() const { return strFile; }
 
+    /** Return whether this database handle is a dummy for testing.
+     * Only to be used at a low level, application should ideally not care
+     * about this.
+     */
+    bool IsDummy() { return env == nullptr; }
```

Comment:
If it's only to be used by low level stuff, maybe friend the function specifically to those classes...